library/src/main/java/io/ona/kujaku/location/clients/AndroidGpsLocationClient.java

```diff
@@ -128,6 +128,9 @@ public void onResult(@NonNull LocationSettingsResult locationSettingsResult) {
     @Nullable
     @Override
     public Location getLastLocation() {
+        if (locationManager == null) {
+            locationManager = (LocationManager) context.getApplicationContext().getSystemService(Context.LOCATION_SERVICE);
```

Comment:
We may need to investigate why location manager is getting null. With this fix we don't know which other methods maybe broken because of this.

Either create a method `locationManager()` that if the references location manager is null reinitializes it so that it can be used in other methods until its known why location manager is becoming null